src/test/java/io/confluent/connect/jdbc/sink/DbStructureTest.java

```diff
@@ -41,6 +60,110 @@ public void testMissingFieldsWithDifferentCase() {
     assertTrue(missingFields(sinkRecords("AaA", "bBb"), columns("aAa", "BbB")).isEmpty());
   }
 
+  @Test (expected = TableAlterOrCreateException.class)
+  public void testMissingTableNoAutoCreate() throws Exception {
+    structure.create(config, connection, tableId,
+        fieldsMetadata);
+  }
+
+  @Test (expected = TableAlterOrCreateException.class)
+  public void testAlterNoAutoEvolve() throws Exception {
+    TableDefinition tableDefinition = mock(TableDefinition.class);
+    when(dbDialect.tableExists(any(), any())).thenReturn(true);
+    when(dbDialect.describeTable(any(), any())).thenReturn(tableDefinition);
+    when(tableDefinition.type()).thenReturn(TableType.VIEW);
+
+    SinkRecordField sinkRecordField = new SinkRecordField(
+        Schema.OPTIONAL_INT32_SCHEMA,
+        "test",
+        false
+    );
+
+    fieldsMetadata = new FieldsMetadata(
+        Collections.emptySet(),
+        Collections.singleton(sinkRecordField.name()),
+        Collections.singletonMap(sinkRecordField.name(), sinkRecordField));
+
+    structure.amendIfNecessary(config, connection, tableId,
+        fieldsMetadata, 5);
+  }
+
+  @Test (expected = TableAlterOrCreateException.class)
+  public void testAlterNotSupported() throws Exception {
+    TableDefinition tableDefinition = mock(TableDefinition.class);
+    when(dbDialect.tableExists(any(), any())).thenReturn(true);
+    when(dbDialect.describeTable(any(), any())).thenReturn(tableDefinition);
+    when(tableDefinition.type()).thenReturn(TableType.VIEW);
+
+    SinkRecordField sinkRecordField = new SinkRecordField(
+        Schema.INT32_SCHEMA,
```

Comment:
Should we make this an optional field just to be sure the call to `DbStructure::amendIfNecessary` fails because the table is a view, not because the schema contains a new non-optional field with no default?

Reply:
Yes, this is a good point just in case.